### ISSUE TITLE
Intel Mac build error caused by meta_parser

### ISSUE THREAD

**Issue by iaomw** [BoomingTech/Pilot#1]
I got build errors for both Xcode and CMake builds. The problem is triggered from **_meta_parser_**. However, it's still malfunction after adding permission with `chmod +x meta_parser`. 

The **_meta_parser_** crashed because `dyld: Library not loaded: @rpath/libclang.dylib`

macOS Bigsur 11.6.5
Xcode 13.2.1 
CMake 3.23

**Comment by Yescafe** [comment]
Change the missing dylib path, this can help you:
https://stackoverflow.com/questions/17703510/dyld-library-not-loaded-reason-image-not-found

**Comment by younif** [comment]
just `cp engine/bin/macOS/libclang.dylib to /usr/local/lib`
it works for me

**Comment by BoomingTech-YuqiaoZhang** [comment]
If you are hasty, you may use [this repository](https://github.com/BoomingTech-YuqiaoZhang/Pilot) instead

**Comment by BoomingTech-YuqiaoZhang** [comment]
You can use the following commands to fix the build error on MacOS temporarily
chmod +x "./engine/bin/MacOS/meta_parse"
export LD_LIBRARY_PATH=./engine/bin/macOS${PATH:+:${PATH}}

**Comment by deepkolos** [comment]
is MacOS 12.2 needed?

<img width="849" alt="image" src="https://user-images.githubusercontent.com/12824616/161565629-ccda47a9-2da5-4e7d-88c0-365701d0520a.png">

**Comment by iaomw** [comment]
> just `cp engine/bin/macOS/libclang.dylib to /usr/local/lib` it works for me

Tested with cp command and manual copy, not working. What is your software version?

**Comment by BoomingTech-YuqiaoZhang** [comment]
> > You can use the following commands to fix the build error on MacOS temporarily chmod +x "./engine/bin/MacOS/meta_parse" export LD_LIBRARY_PATH=./engine/bin/macOS${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
> 
> Thanks, but it's the same result. I may try another solution using `otool` for `libclang.dylib` But ultimately the project script should take care itself.
> 
> https://stackoverflow.com/questions/24045633/dyld-library-not-loaded-error

LB_LIBRARY_PATH may not work, you may use the following install_name_tool instead.
install_name_tool -add-rpath "@executable_path" "./engine/bin/MacOS/meta_parse"

**Comment by Kingfish404** [comment]
macOS 12.3.1 Intel i5 working with fellowing shell command

```
chmod +x "./engine/bin/MacOS/meta_parser"
cmake -S engine -B build -G "Xcode"
cp engine/bin/macOS/libclang.dylib /usr/local/lib
cmake --build build --config "${CMAKE_ARG_BUILD_TYPE_CONFIG}"
rm /usr/local/lib/libclang.dylib
```

then 
```shell
./bin/PilotEditor
```
running.

**Comment by begoss** [comment]
You can use command:
```
cd /.../Pilot/engine/bin/macOS
install_name_tool -add_rpath /.../Pilot/engine/bin/macOS ./meta_parser
```
This works for me.

**Comment by BoomingTech-YuqiaoZhang** [comment]
> libclang.dylib is on the same path as meta_parser, you can use command:
> 
> ```
> cd /.../Pilot/engine/bin/macOS
> install_name_tool -add_rpath /.../Pilot/engine/bin/macOS ./meta_parser
> ```
> 
> This works for me.

The following command is much better:
 install_name_tool -rpath "/Users/hywei/Workspace/metaparser-master/Parser/../3rd_party/LLVM/lib/macOS" '@executable_path' '/Users/HanetakaChou/Documents/github/Pilot/engine/bin/macOS/meta_parser'

**Comment by BoomingTech-YuqiaoZhang** [comment]
> is MacOS 12.2 needed? @BoomingTech-YuqiaoZhang running meta_parser on MacOS 11.6 throw the error
> 
> <img alt="image" width="849" src="https://user-images.githubusercontent.com/12824616/161565629-ccda47a9-2da5-4e7d-88c0-365701d0520a.png">

libc++.1.dylib is missing, I am fixing now

**Comment by BoomingTech-YuqiaoZhang** [comment]
> is MacOS 12.2 needed? @BoomingTech-YuqiaoZhang running meta_parser on MacOS 11.6 throw the error
> 
> <img alt="image" width="849" src="https://user-images.githubusercontent.com/12824616/161565629-ccda47a9-2da5-4e7d-88c0-365701d0520a.png">

using the following commands to fix the libclang.so:
install_name_tool -change '/usr/lib/libc++.1.dylib' 'libc++.1.dylib' '/Users/HanetakaChou/Documents/github/Pilot/engine/bin/macOS/libclang.dylib'
install_name
install_name_tool -rpath '@loader_path/../lib' '@executable_path' '/Users/HanetakaChou/Documents/github/Pilot/engine/bin/macOS/libclang.dylib'

Then, download the missed libc++.1.dylib from the LLVM official web page

**Comment by BoomingTech-YuqiaoZhang** [comment]
> is MacOS 12.2 needed? @BoomingTech-YuqiaoZhang running meta_parser on MacOS 11.6 throw the error
> 
> <img alt="image" width="849" src="https://user-images.githubusercontent.com/12824616/161565629-ccda47a9-2da5-4e7d-88c0-365701d0520a.png">

A merge request is provided. If you are hasty, you may use [this repository](https://github.com/BoomingTech-YuqiaoZhang/Pilot)

**Comment by BoomingTech-YuqiaoZhang** [comment]
> is MacOS 12.2 needed? @BoomingTech-YuqiaoZhang running meta_parser on MacOS 11.6 throw the error
> 
> <img alt="image" width="849" src="https://user-images.githubusercontent.com/12824616/161565629-ccda47a9-2da5-4e7d-88c0-365701d0520a.png">

If you are hasty, you may use [this repository](https://github.com/BoomingTech-YuqiaoZhang/Pilot). Github Workflow shows "build macos" is passing.

**Comment by BoomingTech-YuqiaoZhang** [comment]
A [pull request](https://github.com/BoomingTech/Pilot/pull/13) is provided.

**Comment by forkercat** [comment]
For M1 developers, running these commands works for a temporary fix.

```shell
chmod +x .engine/bin/macOS/meta_parser
sudo cp engine/bin/macOS/libclang.dylib /usr/local/lib
```

**Comment by BoomingTech-YuqiaoZhang** [comment]
> For M1 developers, running these commands works for a temporary fix.
> 
> ```shell
> chmod +x .engine/bin/macOS/meta_parser
> sudo cp engine/bin/macOS/libclang.dylib /usr/local/lib
> ```

A [pull request](https://github.com/BoomingTech/Pilot/pull/13) has been provided. If you are hasty, you may use [this repository](https://github.com/BoomingTech-YuqiaoZhang/Pilot) first. The [github workflow](https://github.com/BoomingTech-YuqiaoZhang/Pilot/actions/workflows/build_linux.yml) showßs "build linux" is passing.

**Comment by BoomingTech-YuqiaoZhang** [comment]
> For M1 developers, running these commands works for a temporary fix.
> 
> ```shell
> chmod +x .engine/bin/macOS/meta_parser
> sudo cp engine/bin/macOS/libclang.dylib /usr/local/lib
> ```

This is NOT recommended. The path "/usr/local/lib" may not exist if the user doesn't use homebrew or macport. 
And the version of the libclang.dylib provided by us may be different from the libclang.dylib provided by homebrew or macport. Some other software may not work properly due to the confliction of the libs.

**Comment by BoomingTech-YuqiaoZhang** [comment]
> just `cp engine/bin/macOS/libclang.dylib to /usr/local/lib` it works for me

This is NOT recommended. The path "/usr/local/lib" may not exist if the user doesn't use homebrew or macport.
And the version of the libclang.dylib provided by us may be different from the libclang.dylib provided by homebrew or macport. Some other software may not work properly due to the confliction of the libs.

**Comment by hyv1001** [comment]
This issue should have been fixed by #53, you can reopen or create a new ticket if it still exists on your machine
